src/backends/simple.ts

```diff
@@ -0,0 +1,442 @@
+    interpolateMode?: InterpolateMode;
+}): Promise<beamcoder.Filterer> => {
+    if (!stream.codecpar.format) {
+        return null;
+    }
+
+    let filterSpec = [`[in0:v]format=${stream.codecpar.format}`];
+
+    if (interpolateFps) {
+        if (interpolateMode === 'high-quality') {
+            filterSpec = [...filterSpec, `minterpolate=fps=${interpolateFps}`];
+        }
+        else if (interpolateMode === 'fast') {
+            filterSpec = [...filterSpec, `fps=${interpolateFps}`];
+        }
+        else {
+            throw new Error(`Unexpected interpolation mode: ${interpolateMode}`);
+        }
+    }
+
+    filterSpec = [...filterSpec];
+
+    const filterSpecStr = filterSpec.join(', ') + '[out0:v]';
+
+    LOG && console.log(`filterSpec: ${filterSpecStr}`);
+
+    return beamcoder.filterer({
+        filterType: 'video',
+        inputParams: [
+            {
+                name: 'in0:v',
+                width: stream.codecpar.width,
+                height: stream.codecpar.height,
+                pixelFormat: stream.codecpar.format,
+                timeBase: stream.time_base,
+                pixelAspect: stream.sample_aspect_ratio,
+            },
+        ],
+        outputParams: [
+            {
+                name: 'out0:v',
+                pixelFormat: outputPixelFormat,
+            },
+        ],
+        filterSpec: filterSpecStr,
+    });
+};
+
+const MAX_PACKET_READS = 1000;
+const STREAM_TYPE_VIDEO = 'video';
+
+/**
+ * A simple extractor that uses beamcoder to extract frames from a video file.
+ */
+export class SimpleExtractor extends BaseExtractor implements Extractor {
+    /**
+     * The demuxer reads the file and outputs packet streams
+     */
+    demuxer: Demuxer = null;
+
+    /**
+     * The decoder reads packets and can output raw frame data
+     */
+    decoder: Decoder = null;
+
+    /**
+     * Packets can be filtered to change colorspace, fps and add various effects. If there are no color space changes or
+     * filters, filter might not be necessary.
+     */
+    filterer: Filterer = null;
+
+    /**
+     * This is where we store our filtered frames closest to the target time. We hang on to them for two reasons:
+     * 1. so we can return them if we get a request for the same time again
+     * 2. so we can return frames close the end of the stream. When such a frame is requested we have to flush (destroy)
+     * the encoder to get the last few frames. This avoids having to re-create an encoder.
+     */
+    filteredFrames: undefined[] | DecodedFrames = [];
+
+    /**
+     * This contains the last raw frames we read from the demuxer. We use it as a starting point for each new query. We
+     * do this ensure we don't skip any frames.
+     */
+    frames = [];
+
+    /**
+     * This contains the last packet we read from the demuxer. We use it as a starting point for each new query. We do
+     * this ensure we don't skip any frames.
+     */
+    packet: Packet = null;
```

Comment:
This should be private
```suggestion
    #packet: Packet = null;
```

Reply:
Yep